### ISSUE TITLE
Review lifecycle of data cached in context.workspaceState

### ISSUE THREAD

**Issue by gjsjohnmurray** [intersystems-community/vscode-objectscript#417]
...
2. The active boolean -- on startup -- has no effect.  Is that desirable?

_Originally posted by @theflyingape in https://github.com/intersystems-community/vscode-objectscript/issues/255#issuecomment-714478618_

Specific example:

Does this caching cause a previous but still valid CSP token to be reused inappropriately?

**Comment by theflyingape** [comment]
No, all the expected CSP token caching is working just fine & appropriately... the point of this was in regards to ObjectScript 'toggle connection to disable' action: it does not appear to do any "CSP logout" to expire/flush the caching.  Any immediate VS Code restart after makes a connection back in without an authentication 'hit' recorded on the server, so I have to presume its reusing the CSP token.

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray Is there anything we need to do here or can this be closed?

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray Here's what I did and the behavior that I saw:

1. Have a server definition with a username stored in plain text but no password anywhere.
2. When prompted for a password, enter it but don't save it in the keychain.
3. Disable the server connection by setting `active` to `false` in `objectscript.conn`.
4. Restart VS Code.
5. Note that you are prompted for a password again (I assume this is the server manager) but the `active` flag is still `false` and the server connection is still disabled
